README.md

```diff
@@ -1,59 +1,104 @@
-            <version>x.y.z</version>
-        </dependency>
-    </dependencies>
+|         Driver Series |         1.6       |   1.5  |   1.4 and bellow  |
+| ---------------------:|:-----------------:|:------:|:-----------------:|
+| Supported Java Runtime| Java 8 and Java 9 | Java 8 | Java 7 and Java 8 |
 
-*Please check the [Releases](https://github.com/neo4j/neo4j-java-driver/releases) for the newest driver version
-available.
 
 
-Connect to a Neo4j 3.0.0+ database:
+### Minimum Viable Snippet
 
-    Driver driver = GraphDatabase.driver( "bolt://localhost:7687", AuthTokens.basic( "neo4j", "neo4j" ) );
-    
-    try ( Session session = driver.session() )
-    {
-        StatementResult rs = session.run( "CREATE (n) RETURN n" );
-    }
-    
-    driver.close();
+The driver is distributed via Maven exclusively.
+To use the driver in your Maven project:
+```html
+<dependency>
+    <groupId>org.neo4j.driver</groupId>
+    <artifactId>neo4j-java-driver</artifactId>
+    <version>x.y.z</version>
+</dependency>
+```
+The version `x.y.x` need to be replaced with a released driver version.
+The latest driver version is always recommended for accessing newest features and recent bug fixes.
+The latest and all available versions of this driver could be found at
+[Maven Central](https://mvnrepository.com/artifact/org.neo4j.driver/neo4j-java-driver) or
+[Releases](https://github.com/neo4j/neo4j-java-driver/releases).
 
-## Building
 
-The source code here reflects the current development status of a new driver version.
-If you want to use the driver in your products, please use the released driver via maven central or check out the
-code with git tags instead.
+To connect to a Neo4j 3.0.0+ database:
+```java
+Driver driver = GraphDatabase.driver( "bolt://localhost:7687", AuthTokens.basic( "neo4j", "PasSW0rd" ) );
+try ( Session session = driver.session() )
+{
+    StatementResult rs = session.run( "CREATE (n) RETURN n" );
+}
+driver.close();
+```
+While using this driver in an application project, we recommend **a single driver** during the whole lifetime of the application
+to benefit from the connection pool maintained in the `driver` object.
 
-### Running tests and creating a package
+The `driver` object holds a connection pool to avoid performance overhead added by establishing TCP connections for each query run.
+Network connections are established on demands when running Cypher queries, and returned back to connection pool after query execution finishes.
+As a result, it is expensive to create and close driver, while super cheap to open and close sessions.
 
-The driver unit tests relies on latest [`boltkit`](https://github.com/neo4j-contrib/boltkit) installed on your local machine. 
-If `boltkit` is not installed, then all tests that requires `boltkit` will be ignored and not be executed.
+The driver is thread-safe, but the session is not thread-safe. So make sure sessions are not passed among threads.
```

Comment:
I think it is fine to safely publish sessions from one thread to another. Users need to make sure `Session` and `Transaction` objects are not used concurrently by multiple threads.